students/etwum/lesson03/mailroom.py

```diff
@@ -0,0 +1,117 @@
+    donor_last_name = input("Please input the last name of the donor. ")
+    donor_full_name = donor_first_name.capitalize() + " " + donor_last_name.capitalize()
+    print('-----------------------------------------------------')
+    print()
+
+    # prints an email if the donors name is currently in the list
+    for x in lstDonors:
+        if x[0] == donor_full_name:
+            email = "Dear {a},\n\nThank you for your generous donation of ${b:.2f} to our charity.\n".format(a=x[0], b=x[1])
+            print(email)
+            break
+
+    # if the donor's name is not in the list it adds the name to the list and ask for a donation amount
+    # then prints an email to the new donor
+    else:
+
+        print("Donor not found")
+        new_donation_amount = None
+        boolValid = False
+        while boolValid != True:
+            try:
+                new_donation_amount = float(input("Please input a donation amount in order to add the donor to the list. "))
+            except ValueError as e:
+                print(e)
+                print("Please input a valid number")
+            else:
+                boolValid = True
+        new_donor.append([donor_full_name, new_donation_amount, 1, new_donation_amount])
+        lstDonors.extend(new_donor)
+        print('-----------------------------------------------------')
+        for x in lstDonors:
+            if x[0] == donor_full_name:
+                email = "Dear {a},\n\nThank you for your generous donation of ${b:.2f} to our charity.\n".format(a=x[0],b=x[1])
+                print(email)
+    print('-----------------------------------------------------')
+
+
+def sort_list(lstDonors):
+    # used to sort the donor list by the total donations column
+
+    return lstDonors[1]
+
+def create_report():
+    # creates a report of the the donors
+
+    for x in lstHeader:
+        print('{:<25}{:<20}{:<17}{:<15}'.format(*x))
+    print("----------------------------------------------------------------------------")
+    for x in sorted(lstDonors,key=(sort_list), reverse= True):
+        print('{:<25} $ {:<20}{:^14} $ {:<15}'.format(*x))
+
+
+print("Welcome to the Charity Mail Room")
+print("------------------------------------------------------------------------")
+# ****Input/Output****
+while True:
+    # Option menu
+
+    print("""
+    Menu of Options
+    1) Send Thank You Note
+    2) Create Report
+    3) Exit Program
+    """)
+    strChoice = None
+
+    # try/except block to make sure the user inputs a valid option
+    try:
+        strChoice = int(input("Which option would you like to perform? Input a number [1 to 3] "))
+        if strChoice == 1 or strChoice == 2 or strChoice == 3:
+            print()
+        else:
+            raise Exception
+    except Exception:
+        print("Please input a valid option: 1, 2, or 3")
+
+    if (strChoice == 1):
+        send_thank_you()
+        continue
+
+    elif (strChoice == 2):
+        create_report()
+        continue
```

Comment:
no need for continue statements here and above, your program will continue naturally